File: modules/coupon-domain/src/main/java/com/huythanh0x/udemycoupons/dto/VideoSourceDTO.java
package com.huythanh0x.udemycoupons.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for a video source (MP4 file at specific resolution).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VideoSourceDTO {
    private String type; // e.g., "video/mp4"
    private String label; // e.g., "720", "480", "360", "144", "Auto"
    private String file; // URL to the video file
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/dto/VideoSourceDTO.kt
package com.thanh0x.coursedeal.dto

/**
 * DTO for a video source (MP4 file at specific resolution).
 */
data class VideoSourceDTO(
    val type: String? = null, // e.g., "video/mp4"
    val label: String? = null, // e.g., "720", "480", "360", "144", "Auto"
    val file: String? = null // URL to the video file
)
